shared/src/main/scala/mlscript/codegen/Symbol.scala

```diff
@@ -0,0 +1,158 @@
+package mlscript.codegen
+
+import mlscript.utils.shorthands._
+import mlscript.Type
+import mlscript.JSClassDecl
+import mlscript.MethodDef
+import mlscript.Term
+import mlscript.TypeName
+
+sealed trait LexicalSymbol {
+
+  /**
+    * The lexical name of the symbol. This is different from the runtime name,
+    * the name of the symbol in the generated code. We allow duplicates lexical
+    * names in the same scope.
+    */
+  def lexicalName: Str
+}
+
+sealed trait RuntimeSymbol extends LexicalSymbol {
+  def runtimeName: Str
+}
+
+sealed trait TypeSymbol extends LexicalSymbol {
+  val params: Ls[Str]
+  val actualType: Type
```

Comment:
These two params seem unused. Remove them? Or justify why we'll need them in the future.

Reply:
I noticed that @twitu is using two scopes (one for types and another for terms) in TypeScript code generation. So preserving `TypeSymbol` and related information could simplify the workflow there.

Reply:
Ok, though he's not using symbols yet because they weren't really available. we'll look into using that info.

Reply:
OK. So I'm keeping it. We can remove them if it turns out to be useless.